plugins/catalog-backend/src/next/processing/ProcessorCacheManager.ts

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { JsonObject, JsonValue } from '@backstage/config';
+import { CatalogProcessor } from '../../ingestion/processors';
+import { CatalogProcessorCache } from '../../ingestion/processors/types';
+import { isObject } from './util';
+
+class SingleProcessorSubCache implements CatalogProcessorCache {
```

Comment:
this hierarchy just makes me think "there's gotta be a simpler way" :) like, each level being one single class that has the ability to descend one key down - just checking, do i not yet have that key, then create an empty object at that key and construct a new instance of myself on whatever the new "root" at that key is